File: all-submissions/0159-longest-substring-with-at-most-two-distinct-characters/solution.java
class Solution {
    public int lengthOfLongestSubstringTwoDistinct(String s) {
        Map<Character, Integer> myMap = new HashMap<>();
        int l = 0;
        int r = 0;
        int len = 0;
        int subStrSize = 0;
        int count = 0;
        while(r < s.length()){
            char rChar = s.charAt(r);
            myMap.put(rChar, myMap.getOrDefault(rChar,0)+1);
            while(myMap.size() > 2) {
                char lChar = s.charAt(l);
                if (myMap.get(lChar) > 1){
                    myMap.put(lChar, myMap.get(lChar)-1);
                } else {
                    myMap.remove(lChar);
                }
                subStrSize--;
                l++;
            } 
            subStrSize++;
            count = Math.max(count, subStrSize);
            r++;
        }
        return count;
        
    }
}

File: all-submissions/0159-longest-substring-with-at-most-two-distinct-characters/solution.py
class Solution:
    def lengthOfLongestSubstringTwoDistinct(self, s: str) -> int:
        char_map = {}
        left = 0
        right = 0
        maxLength = 0
        while(right<len(s)):
            char_map[s[right]] = char_map.get(s[right], 0)+1
            while len(char_map) > 2:
                char_map[s[left]] -=1
                if char_map[s[left]] == 0:
                    del char_map[s[left]]
                left +=1
            maxLength = max(maxLength, right-left+1)
            right +=1
        return maxLength

